### PR TITLE
Adding template for CODEOWNERS for DellEmc Enterprise SONiC ansible c…

### PR DESCRIPTION
**Pull request by kannankvs**
##### SUMMARY
Adding template for CODEOWNERS for DellEmc Enterprise SONiC ansible collections.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
Template for CODEOWNERS

### CODE REVIEW

**Comment by kerry-meyer** [comment]
There are a few warnings to be addressed in this file. I have posted inline comments for two of them.

For the other warning:

```
Run command with data: /root/.ansible/test/venv/sanity/3.11/4f53cda1/bin/python /root/ansible/test/lib/ansible_test/_util/controller/sanity/code-smell/line-endings.py
ERROR: Found 1 line-endings issue(s) which need to be resolved:
ERROR: .github/CODEOWNERS:0:0: use "\n" for line endings instead of "\r\n"
See documentation for help: https://docs.ansible.com/ansible-core/devel/dev_guide/testing/sanity/line-endings.html
```

It looks like this file must have been created on a Windows PC. The line ending sequence (CR/LF) is not allowed by the Ansible sanity checker. A Unix line ending sequence (LF) is required. Please convert the file to a "Unix" file. You can do this by copying the file (e.g. via "scp" or "pscp" or just copy and paste) to a Unix/Linux server file and executing "dos2unix <filename>" on the Unix/Linux system where you have copied the file.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# CODEOWNERS
```
Comment (kerry-meyer) [inline .github/CODEOWNERS:1]: I think this line is causing one of the Ansible sanity errors encountered when the PR was submitted. It seems unnecessary. Please try deleting this line and the following line to see if it fixes the sanity check problem:

```
Running sanity test "shebang"
Run command with data: /root/.ansible/test/venv/sanity/3.11/4f53cda1/bin/python /root/ansible/test/lib/ansible_test/_util/controller/sanity/code-smell/shebang.py
ERROR: Found 1 shebang issue(s) which need to be resolved:
ERROR: .github/CODEOWNERS:0:0: file without shebang should not be executable
See documentation for help: https://docs.ansible.com/ansible-core/devel/dev_guide/testing/sanity/shebang.html
```

You can see the sanity output from which the above excerpt was quoted by clicking on the "details" link next to a failing sanity test shown at the bottom of the PR.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# CODEOWNERS
+#
+# documentation for this file can be found at:
+# https://help.github.com/en/articles/about-code-owners
+
+# These are the default owners for the code and will
+# be requested for review when someone opens a pull request.
+# order is alphabetical for easier maintenance.
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply .github/CODEOWNERS]: The above comment on "dos2unix <filename>" has been resolved as suggested. The above comment on adding "The" at the beginning has been addressed as well.

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+# These are the default owners for the code and will
+# be requested for review when someone opens a pull request.
+# order is alphabetical for easier maintenance.
+#
+# Kerry Meyer (kerry_meyer@dell.com)
+# Mingjun Zhang (Mingjun_Zhang@Dell.com)
+# Shade Talabi (Shade_Talabi@Dell.com)
+
+# for all files:
+* @kerry-meyer @mingjunzhang2019 @stalabi1
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply .github/CODEOWNERS]: We have removed the Mingjun id from the list as suggested.

File: .github/CODEOWNERS (hunk 1 above)
Comment (kannankvs) [reply .github/CODEOWNERS]: We were able to identify the issue as shebang issue and we are working on the same to fix it.

Comment (kerry-meyer) [reply .github/CODEOWNERS]: You are correct that it is a failure for the shebang check. My initial comment attributing the cause of the problem to the first line was wrong. Instead, it is an incorrect file mask for the CODEOWNERS file. The "x" ("executable") bit should be off for "user", "group", and "other". 

e.g.

cd .github
chmod 644 CODEOWNERS

Then push the modified file to your branch.

See below for more details.

------------------------
Based on reading the Ansible source code for the shebang sanity check, ( https://github.com/ansible/ansible/blob/devel/test/lib/ansible_test/_util/controller/sanity/code-smell/shebang.py )  I think the problem is that the ".github/CODEOWNERS file has an invalid file mask. 


```
    for path in sys.argv[1:] or sys.stdin.read().splitlines():
        with open(path, 'rb') as path_fd:
            shebang = path_fd.readline().strip()
            mode = os.stat(path).st_mode
            executable = (stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH) & mode

            if not shebang or not shebang.startswith(b'#!'):
                if executable:
                    print('%s:%d:%d: file without shebang should not be executable' % (path, 0, 0))
```

REF: https://docs.python.org/3/library/stat.html  (for IXUSR, IXGRP, and IXOTH definitions)

To fix this, turn off the "execute" bits for all users in the copy of your file in your workspace and push the modified file. Use "664" file mask for this file.

e.g.

cd .github
chmod 644 CODEOWNERS


************

Looking at the current Unix file permissions set for this file on your branch:

```
kerry@kerry-ansible:~/garb_kannan_codereviewers$ git clone https://github.com/kannankvs/dellemc.enterprise_sonic
Cloning into 'dellemc.enterprise_sonic'...
remote: Enumerating objects: 2623, done.
remote: Counting objects: 100% (919/919), done.
remote: Compressing objects: 100% (558/558), done.
remote: Total 2623 (delta 562), reused 561 (delta 318), pack-reused 1704
Receiving objects: 100% (2623/2623), 932.13 KiB | 3.12 MiB/s, done.
Resolving deltas: 100% (1253/1253), done.
kerry@kerry-ansible:~/garb_kannan_codereviewers$ cd dellemc.enterprise_sonic/
kerry@kerry-ansible:~/garb_kannan_codereviewers/dellemc.enterprise_sonic$ git checkout code_owners_template
Branch 'code_owners_template' set up to track remote branch 'code_owners_template' from 'origin'.
Switched to a new branch 'code_owners_template'
kerry@kerry-ansible:~/garb_kannan_codereviewers/dellemc.enterprise_sonic$ git branch
* code_owners_template
  main
kerry@kerry-ansible:~/garb_kannan_codereviewers/dellemc.enterprise_sonic$ cd .github
kerry@kerry-ansible:~/garb_kannan_codereviewers/dellemc.enterprise_sonic/.github$ ls -ld *
-rwxrwxr-x 1 kerry kerry  406 Oct  6 15:28 CODEOWNERS   <<< The "x" bit is turned on for "user", "group", and "other".
drwxrwxr-x 2 kerry kerry 4096 Oct  6 15:24 workflows      
```

File: .github/CODEOWNERS (hunk 3 above)
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply .github/CODEOWNERS]: Thanks. I am resolving this issue.

File: .github/CODEOWNERS (hunk 2 above)
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply .github/CODEOWNERS]: Thank you. This issue is resolved.

File: .github/CODEOWNERS (hunk 1 above)
Comment (kerry-meyer) [reply .github/CODEOWNERS]: I see that a commit has been done to fix the file permissions mask.

I will mark this issue "Resolved".

**Review by kerry-meyer — APPROVED** [review]
All issues have been addressed and resolved.

Thank you for providing this template and for making the additional changes needed to resolve all review issues.

I will go ahead with approving and merging this PR.

**Pull request merged**